src/ui/components/DeckBuilder/useDeckBuilder.ts

```diff
@@ -0,0 +1,72 @@
+import { useCallback, useMemo, useState } from 'react'
+
+import * as AllCards from '../../../game/cards'
+import { DECK_SIZE } from '../../../game/config'
+import { pricing } from '../../../game/services/Pricing'
+import { CardType, ICard, ICrop } from '../../../game/types'
+import { isCard } from '../../../game/types/guards'
+
+interface UseDeckBuilderProps {
+  onDone: (deck: Map<ICard, number>) => void
+}
+
+const sortedCards = (() => {
+  const cards = (Object.values(AllCards) as unknown[]).filter(isCard)
+
+  const crops = cards.filter(c => c.type === CardType.CROP) as ICrop[]
+  const water = cards.filter(c => c.type === CardType.WATER)
+  const tools = cards.filter(c => c.type === CardType.TOOL)
+  const events = cards.filter(c => c.type === CardType.EVENT)
```

Comment:
Use the `isXCardInstance` (like `isCropCardInstance` and `isWaterCardInstance`) utility functions from `src/game/types/index.ts` instead of using custom sort functions here.

Also, move these definitions to `src/game/cards/index.ts` in case we want to reuse them later.